packages/calling-stateful-client/src/CallAgentDeclarative.ts

```diff
@@ -209,8 +209,20 @@ class ProxyCallAgent implements ProxyHandler<DeclarativeCallAgent> {
         };
       }
       case 'dispose': {
+        // Wrapping CallAgent.dispose in a callback type (): Promise<void> to accomodate the change of CallAgent.dispose
+        // in calling beta version 1.8.0-beta.1 from callback type (): Promise<void> to (): void
+        const disposeCallbackWrapper = (): Promise<void> => {
+          target.dispose();
+          return Promise.resolve();
+        };
+        /* @conditional-compile-remove(calling-beta-sdk) */
+        return (): void => {
+          disposeCallbackWrapper().then(() => {
```

Comment:
Why aren't we using await/async here?

Reply:
Good catch. I changed the beta code to be a simple synchronous call of CallAgent.dispose.